rawnode.go

```diff
@@ -194,14 +194,22 @@ func MustSync(st, prevst pb.HardState, entsnum int) bool {
 	return entsnum != 0 || st.Vote != prevst.Vote || st.Term != prevst.Term
 }
 
-func needStorageAppend(rd Ready, haveMsgsAfterAppend bool) bool {
+func needStorageAppendMsg(r *raft, rd Ready) bool {
```

Comment:
This isn't the right place for this comment but I didn't want to post it in the main thread (no "Resolve" feature there):

I wonder if there's a simple UX improvement we could make to ready handling when using async writes. Right now, there are a bunch of fields that are unused in that case:

- MustSync
- CommittedEntries
- Entries
- Snapshot
- any others?

What about a helper method

```go
func AsyncStorageWritesReady(rd Ready) AsyncReady {
  // ...
}
```

that basically returns only the things you need to care about as a user of the async interface? This would also provide a convenient place to anchor documentation without having to interleave the two different usage patterns.

Reply:
I like this idea. Let me try it out and see how it looks.